### ISSUE TITLE
[HOLD for payment 2023-05-04] [$1000] After a login, if the user goes offline, their Personal Details are not loaded

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16985]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app and logout if you're currently logged in
2. Log in
3. Close internet connection or use force offline to mimic offline mode/disable WiFi or data to go offline
4. Open Settings
5. Open Profile
6. Open Personal Details and observe that the data does not load/is not showing

## Expected Result: 
App should load the Personal Details even if the user is offline.

## Actual Result: 
When you login, App does not load Personal Details data. Please also refer to this comment - https://github.com/Expensify/App/issues/16985#issuecomment-1498308394

When you're offline and then enable WiFi/cell data again, and you're already on the "Personal Details" screen, the data also doesn't repopulate. You must exit that's screen and enter it again to see your personal data populate.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.95-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/230236784-14e65abf-c01f-44c3-9be0-60706debf2f8.mp4

https://user-images.githubusercontent.com/43996225/230236809-9b103276-2062-42b6-8f07-14012479ba13.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1680689788347159
 
[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b574a3b6c95ac76c</li>
        <li>Upwork Job ID: 1643770433495482368</li>
        <li>Last Price Increase: 2023-04-13</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @jliexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by jliexpensify** [comment]
Can repro on v95 + Pixel 3a (Android). 

Another behaviour I noticed: when you enable WiFi/cell data again, and you're on the "Personal Details" screen, the data doesn't repopulate. You must exit that's screen and enter it again to see your personal data populate.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01b574a3b6c95ac76c

**Comment by MelvinBot** [comment]
Current assignee @jliexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @0xmiroslav (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @deetergp (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by MelvinBot** [comment]
📣 @blockdev723! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by blockdev723** [comment]
Greetings,
I've checked the issue and understood 100%.
My solution to fix this issue is to save the profile detail information on the local storage or Redux.
I will complete this task in 1 day.
I am looking forward to hearing from you.
Kind regards, Daniil. from Upwork.

Contributor details
Your Expensify account email: d842443@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01b5053339c1ddbe12

**Comment by MelvinBot** [comment]
✅ Contributor details stored successfully. Thank you for contributing to Expensify!

**Comment by ImBIOS** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Personal details are not showing any data if we offline when we never open it while online.

### What is the root cause of that problem?

`API.read('OpenPersonalDetailsPage');` this API call is not called and cached with onyx on login

### What changes do you think we should make in order to solve the problem?


We need to call and cache this API on user log-in, or specifically when the user is redirected from the login to the report screen.

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief, and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

_Nope_

**Comment by Pujan92** [comment]
API Call to get personal details we are making when a user comes to that route(`settings/profile/personal-details`). Initially, on fresh login we don't have the details. IMO this is not a bug. 

> Another behaviour I noticed: when you're offline and then enable WiFi/cell data again, and you're on the "Personal Details" screen, the data doesn't repopulate. You must exit that's screen and enter it again to see your personal data populate.

For above point of @jliexpensify we can make the write api request instead of read which will add this call in queue and get executed once the network is online.

**Comment by dhanashree-sawant** [comment]
Hi @Pujan92, if we are making only that specific API call on page visit, wouldn't that be the main cause for the issue? Is there any specific reason we aren't calling that API to fetch data on login?

**Comment by blockdev723** [comment]
Soution 1: 
When getting profile information, we must fetch personal detail information together.

Solution 2:
if the detail information is empty, then detect the Network On state on the detail page. Once the network is on, then fetch the information again. By doing so, without going back, and opening the page again, this issue can be fixed.

**Comment by Pujan92** [comment]
> Is there any specific reason we aren't calling that API to fetch data on login?

I am not sure but might be to avoid unnecessary calls after login.

**Comment by dhanashree-sawant** [comment]
Okay if it doesn't have any specific reason, I think we should still fix it?

**Comment by nemapranay** [comment]
Hey,

From offline point of view we can have a seperate class to handle it on top of app screens so that whenever we go offline we can have option to restart from there directly,
and  solution for profile data not coming is - whenever we are calling login then in response itself we should send profile data and that can be store into redux or local storage that will be more helpful.
so there is no need to call fetch api on page visit so it will solve your other problems too.

I can solve it in 2 days with PR raised 

my upwork profile is - https://www.upwork.com/freelancers/~016393179a3bf5a1ca 

Thanks

**Comment by MelvinBot** [comment]
📣 @nemapranay! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by waseem-mansha-gondal** [comment]
Here is my solution : 

We need to call this on after login successfully  
We need to call below function  PersonalDetails.openPersonalDetailsPage();
and it will call API.read('OpenPersonalDetailsPage');
I will take 1 day of development  

Contributor details
Your Expensify account email: waseem.mansha@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01f5769482e70c2260?viewMode=1

**Comment by MelvinBot** [comment]
⚠️ Invalid email. Please make sure to create an Expensify account with this email first [here](https://new.expensify.com/).

**Comment by waseem-mansha-gondal** [comment]
I have created an account on Expensify as well with this email.

**Comment by MelvinBot** [comment]
📣 @waseem-mansha-gondal! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by tienifr** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
Personal details are not showing data if user goes offline after login

### What is the root cause of that problem?
In `PersonalDetailsInitialPage.js` we call `PersonalDetails.openPersonalDetailsPage()` to get personal data. If user goes offline after login, we can't get data at that time. Beside we use `API.read('OpenPersonalDetailsPage')` that why we don't have logic to re-call the failed API when user goes online

### What changes do you think we should make in order to solve the problem?

Solution 1. We should fix it on BE
When we call 'OpenApp' API, BE should return the personalDetail data. Then we can remove `PersonalDetails.openPersonalDetailsPage();` in `PersonalDetailsInitialPage.js`
Solution 2. We can call `PersonalDetails.openPersonalDetailsPage()` after this line https://github.com/Expensify/App/blob/78a9b540f6244276599d58f9aa3fc7d671441c8d/src/libs/Navigation/AppNavigator/AuthScreens.js#L117
Solution 3. If it's not a bug, we can improve by moving `PersonalDetails.openPersonalDetailsPage()` in `PersonalDetailsInitialPage.js` to componentDidMount. And we can use API.write('OpenPersonalDetailsPage') to get the personalDetail data, so when users go online they can see the data.

**Comment by tienifr** [comment]
updated the proposal a little bit.

**Comment by vladbogza** [comment]
### Proposal

### What is the root cause of that problem?
The root issue here is that a users' `Personal Details` are never loaded in any API response except for when the `OpenPersonalDetailsPage` command is called(that is only called when a user clicks on the `Personal Details` button). 

There are just two endpoints being called on page load. 
<img width="755" alt="Screen Shot 2023-04-07 at 9 45 39 PM" src="https://user-images.githubusercontent.com/114418542/230697935-c0bc67c7-1aa9-41f3-8afd-68e8ea05c3ff.png">

1. `AuthenticatePusher` <- Authentication
2. `OpenReport` <- loading the user report. The attached file is an example of the response. 
[doc.txt](https://github.com/Expensify/App/files/11182263/doc.txt)


In neither of those calls are the Personal Details returned. No data returned means no data to store. 

### What changes do you think we should make in order to solve the problem?
Solution 1: Modify the `OpenReport` call to include the Personal details. This is the optimal solution. Data returned means data can be stored. Otherwise it doesn't exist until the call to get personal details are made. 

Not recommended Solution 2(suboptimal): Call the `OpenPersonalDetailsPage ` endpoint on load and store it locally. 



Contributor details
Your Expensify account email: [vladimirbogza@gmail.com](mailto:vladimirbogza@gmail.com)	
Upwork Profile Link: https://www.upwork.com/freelancers/~0117583cb41393c914

**Comment by MelvinBot** [comment]
⚠️ Missing/invalid email or upwork profile link. Please make sure you add both your Expensify email and Upwork profile link in the format specified.

**Comment by situchan** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
data not refreshed on Personal Details page even when user goes online

### What is the root cause of that problem?
data is fetched from `PersonalDetails.openPersonalDetailsPage` but this doesn't get called when user goes online

### What changes do you think we should make in order to solve the problem?
call that api on both cases - component mount, network changes from offline to online
We can use similar approach to `MoneyRequestModal` so add `props.network.isOffline` as dependency in `useEffect` and call this api inside effect but only when that dependency value is false

Improvement:
I found wrong logic on that component. `PersonalDetails.openPersonalDetailsPage()` is called at the start of functional component with any `useEffect` or `useCallback`. This makes call the api every time state or prop value is updated.
i.e. when user updates any profile info on another device, `PRIVATE_PERSONAL_DETAILS` onyx value is updated and component is re-rendered. So we should move this call to inside effect.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @lschurr (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by deetergp** [comment]
> "When you're offline and then enable WiFi/cell data again, and you're already on the "Personal Details" screen, the data also doesn't repopulate. You must exit that's screen and enter it again to see your personal data populate." - this is actual bug to be fixed I think.

@0xmiroslav I agree 100%. Let's focus on fixing this and consider not having personal details loaded prior to having visited the page while connected to the internet as _working as intended_.

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by lschurr** [comment]
Are we still waiting on proposals @0xmiroslav?

**Comment by tienifr** [comment]
@0xmiroslav Do you review my proposal https://github.com/Expensify/App/issues/16985#issuecomment-1498718995? Thanks

**Comment by tienifr** [comment]
@0xmiroslav 
> Solution 3. If it's not a bug, we can improve by moving PersonalDetails.openPersonalDetailsPage() in PersonalDetailsInitialPage.js to componentDidMount. And we can use API.write('OpenPersonalDetailsPage') to get the personalDetail data, so when users go online they can see the data.
I think my third solution can fix this issue. Sorry for using componentDidMount, I thought it's class component, I mean useEffect with empty dependency.
And although the improvements are not mentioned on the requirement but we can easily notice them when perform this issue. when we create the PR, we have to make sure there is no console errors right? 
Pardon me if I'm wrong something. Thanks 😊

**Comment by tienifr** [comment]
https://user-images.githubusercontent.com/113963320/232547568-660cd296-15f3-4750-9024-eb9f9dc17de5.mp4

Here is my result and if the `OpenPersonalDetailsPage` API is called we don't need to call it again when users change network status many times (offline to online) by using API.write() as other places. @0xmiroslav Can you help review and please let me know if I miss something

**Comment by MelvinBot** [comment]
@deetergp @lschurr @0xmiroslav this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by lschurr** [comment]
Any updates @deetergp @0xmiroslav?

**Comment by deetergp** [comment]
Apologies @0xmiroslav & @lschurr. I too like @situchan's proposal and am confused about @tienifr's comments, could someone clarify? Are they making an updated proposal or proposing a separate issue that needs to be fixed. Thanks!

**Comment by tienifr** [comment]
@deetergp  My proposal is to move PersonalDetails.openPersonalDetailsPage() to useEffect:
useEffect(()=>{
PersonalDetails.openPersonalDetailsPage()
},[])

How to re-fetch the personal data when users go online? 
I use API.write('OpenPersonalDetailPage') instead of API.read, so the failed API will re-call when users go online again.
Why do I think it's better than @situchan's proposal? Let's imagine users are in personal detail page and change network status many times => we don't need to call this API again if isOffline is false.

**Comment by tienifr** [comment]
@deetergp I also find some error console when perform this bug, so I think it should be fixed. If you think it's not related to this GH, please ignore it 😁

**Comment by situchan** [comment]
> Let's imagine users are in personal detail page and change network status many times => we don't need to call this API again if isOffline is false.

This is no bug but expected. It's normal to sync data after become online. Why user change network status many times?

**Comment by bernhardoj** [comment]
We already had a discussion about `reportActions` console error before here https://expensify.slack.com/archives/C01GTK53T8Q/p1679985742110289

**Comment by tienifr** [comment]
@situchan 
> This is no bug but expected. It's normal to sync data after become online. Why user change network status many times?


I agree there is no bug but we can improve it by avoiding call API many times like other places also use API.write. We don't need to sync data after become online because we're updating the personal information through pusher

**Comment by situchan** [comment]
> other places also use API.write

they're really write apis. do you find any case where "open page" api is write?

**Comment by tienifr** [comment]
@situchan @deetergp  In this case, why do we need to recall the `OpenPersonalDetailsPage` API after become online? When users go online, `ReconnectApp` API is called and it's result already includes personalDetail data.

**Comment by situchan** [comment]
> @situchan @deetergp In this case, why do we need to recall the `OpenPersonalDetailsPage` API after become online? When users go online, `ReconnectApp` API is called and it's result already includes personalDetail data.

`ReconnectApp` doesn't have full personalDetails data including address, legalFirstName, legalLastName, dob. That's why this issue was reported.

**Comment by MelvinBot** [comment]
@deetergp, @lschurr, @0xmiroslav Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by tienifr** [comment]
@situchan Did you test the case users open personal detail page, go offline, update personal detail from other device and go online again? I can confirm the personal detail data is updated that why we don't need to call OpenPersonalDetaiPage API again

**Comment by deetergp** [comment]
Let's go with @situchan's proposal.

**Comment by situchan** [comment]
Thanks @deetergp 
Can you please officially assign me to the issue?

**Comment by MelvinBot** [comment]
📣 @situchan You have been assigned to this job by @lschurr!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.6-0 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17944


If no regressions arise, payment will be issued on 2023-05-04. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@0xmiroslav] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@0xmiroslav] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@0xmiroslav] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@0xmiroslav] Determine if we should create a regression test for this bug.
- [ ] [@0xmiroslav] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@lschurr] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by lschurr** [comment]
@dhanashree-sawant @situchan @0xmiroslav - could you apply for the job in Upwork? https://www.upwork.com/jobs/~01b574a3b6c95ac76c

**Comment by dhanashree-sawant** [comment]
Thanks @lschurr, I have applied to the upwork job

**Comment by lschurr** [comment]
@0xmiroslav - do we need a regression test for this one?
